homeassistant/components/ads.py

```diff
@@ -0,0 +1,207 @@
+"""
+ADS Component.
+
+For more details about this component, please refer to the documentation.
+https://home-assistant.io/components/ads/
+
+"""
+import threading
+import struct
+import logging
+import ctypes
+from collections import namedtuple
+import voluptuous as vol
+from homeassistant.const import CONF_DEVICE, CONF_PORT, CONF_IP_ADDRESS, \
+    EVENT_HOMEASSISTANT_STOP
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pyads==2.2.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_ADS = 'data_ads'
+
+# Supported Types
+ADSTYPE_INT = 'int'
+ADSTYPE_UINT = 'uint'
+ADSTYPE_BYTE = 'byte'
+ADSTYPE_BOOL = 'bool'
+
+
+ADS_PLATFORMS = ['switch', 'binary_sensor', 'light']
+DOMAIN = 'ads'
+
+# config variable names
+CONF_ADSVAR = 'adsvar'
+CONF_ADSTYPE = 'adstype'
+CONF_ADS_USE_NOTIFY = 'use_notify'
+CONF_ADS_POLL_INTERVAL = 'poll_interval'
+CONF_ADS_FACTOR = 'factor'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICE): cv.string,
+        vol.Required(CONF_PORT): cv.port,
+        vol.Optional(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_ADS_POLL_INTERVAL, default=1000): cv.positive_int,
+        vol.Optional(CONF_ADS_USE_NOTIFY, default=True): cv.boolean,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    import pyads
+    """ Set up the ADS component. """
```

Comment:
All docstrings needs to conform with [PEP257](https://www.python.org/dev/peps/pep-0257/) and need to be present.